### PR TITLE
harness/openssl: multiple OpenSSL builds

### PR DESCRIPTION
**Pull request by woodruffw**
~~Not integrated anywhere yet.~~

### CODE REVIEW

**Comment by woodruffw** [comment]
This now works. It's a bit hard-coded in places (and is incredibly slow on ARM based Macs, because of nested emulation/virtualization), but performs reasonably and generates results for each enumerated version of OpenSSL.

File: Makefile [hunk 1]
```diff
@@ -80,8 +80,11 @@ test-go:
 
 .PHONY: test-openssl
 test-openssl:
-	$(MAKE) -C harness/openssl
-	$(MAKE) run ARGS="harness ./harness/openssl/main --output ./results/openssl.json"
+	$(MAKE) -C harness/openssl openssl-1.1.1 openssl-3.0 openssl-3.1 openssl-3.2
+	$(MAKE) run ARGS="harness --output ./results/openssl-1.1.1.json -- docker run --platform linux/amd64 --rm -i x509-limbo-openssl-1.1.1"
+	$(MAKE) run ARGS="harness --output ./results/openssl-3.0.json -- docker run --platform linux/amd64 --rm -i x509-limbo-openssl-3.0"
+	$(MAKE) run ARGS="harness --output ./results/openssl-3.1.json -- docker run --platform linux/amd64 --rm -i x509-limbo-openssl-3.1"
+	$(MAKE) run ARGS="harness --output ./results/openssl-3.2.json -- docker run --platform linux/amd64 --rm -i x509-limbo-openssl-3.2"
```
Comment (woodruffw) [inline Makefile:87]: I should really dedupe this. Not sure how best to, yet.

**Review by alex — COMMENTED** [review]
Looks fine. I'm concerned about the maintainability of the external docker images.

**Comment by woodruffw** [comment]
> Looks fine. I'm concerned about the maintainability of the external docker images.

Yeah, it's not ideal. I put them on GHCR under my own namespace for convenience, but ideally there would be pre-existing ones we could reuse that I'm not building from scratch. 

(Maybe they belong in this repo too?)

**Comment by alex** [comment]
It's also not crazy to just build them in CI and use caching for
performance. That's what we do in pyca/cryptography.

On Thu, Feb 22, 2024 at 12:55 PM William Woodruff
***@***.***> wrote:
>
> Looks fine. I'm concerned about the maintainability of the external docker images.
>
> Yeah, it's not ideal. I put them on GHCR under my own namespace for convenience, but ideally there would be pre-existing ones we could reuse that I'm not building from scratch.
>
> (Maybe they belong in this repo too?)
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because your review was requested.Message ID: ***@***.***>



-- 
All that is necessary for evil to succeed is for good people to do nothing.

**Comment by woodruffw** [comment]
> It's also not crazy to just build them in CI and use caching for performance. That's what we do in pyca/cryptography.

Sounds good -- I'm going to merge as-is (to help me continue triage on #204), and follow-up with some non-Docker builds.

**Pull request merged**